src/algebra/monoid_algebra/basic.lean

```diff
@@ -716,6 +716,58 @@ finset.induction_on s rfl $ λ a s has ih, by rw [prod_insert has, ih,
 
 end
 
+/-- Given a map `• : G × M → M` for `M` a `k`-module, the map `k[G] × M → M` sending
+`(Σ kᵢgᵢ, m) ↦ Σ kᵢ• (gᵢ • m)`. -/
+instance total_scalar {k : Type*} [semiring k] {G : Type*} {M : Type*}
+  [add_comm_monoid M] [has_scalar G M] [module k M] :
+  has_scalar (monoid_algebra k G) M :=
```

Comment:
Does the action this forms when `k=R`, `G=R` and `M=monoid_algebra R R` agree with `has_mul.to_scalar`? If not, this forms a diamond, and is not really safe as a global instance.

Reply:
If you make the instance only be on `group_ring` then my concern isn't relevant.

Reply:
Ahh, no, they don't agree. They've been defs before with no issue and are quite general; could I leave them where they are and make them defs, or should I just move them to group_ring?

Reply:
Making them defs in this file should be fine too, although make sure to add a comment explaining the conflicts that would occur if these were instances.